packages/mongo-nav/src/mongo-select/ProjectSelect.tsx

```diff
@@ -112,6 +120,29 @@ function ProjectSelect({
     }
   };
 
+  const fetchProjectsAsGlobalUser = (searchTerm: string) => {
+    const queryString = searchTerm ? `?term=${searchTerm}` : '';
+    const endpointURI = `${hosts.cloud}/user/shared/projects/search${queryString}`;
+    return fetch(endpointURI, {
+      credentials: 'include',
+      mode: 'cors',
+      method: 'GET',
+    });
+  };
+
+  const filterDataAsAdmin = debounce(() => {
+    fetchProjectsAsGlobalUser(value)
+      .then(response => response.json())
+      .then(data =>
+        data.map(({ cid, gn }: ProjectFilterResponse) => ({
```

Comment:
Just a thought for potential improvement here – you could reassign these variables to the desired name in the function parameters to give them more meaning here:

```ts
.then(data => data.map(({ cid: projectId, gn: projectName }: ProjectFilterResponse) => ({ projectId, projectName }))
```

One more thought is that it may be more readable to move this map function (and the similar one in I believe the org select) to its own variable:

```ts
const filterDataAsAdmin = debounce(() => {
    const processData = ({
        cid: projectId,
        gn: projectName,
    }: ProjectFilterResponse) => ({ projectId, projectName });

    fetchProjectsAsGlobalUser(value)
      .then(response => response.json())
      .then(processData)
      .then(setFilteredData)
      .catch(console.error);
});

```

Reply:
Sure thing! I think that will be cleaner as well since there is so much chaining going on.